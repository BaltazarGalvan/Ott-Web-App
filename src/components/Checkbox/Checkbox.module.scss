@use '../../styles/variables';
@use '../../styles/theme';

.checkbox {
  label {
    margin-left: variables.$base-spacing / 2;
    font-family: theme.$body-font-family;
    font-size: 14px;
    cursor: pointer;

    > a {
      color: variables.$white;
      font-weight: 700;
      text-decoration: none;

      &:hover {
        text-decoration: underline;
      }
    }
  }

  .header {
    margin-bottom: 8px;
    font-family: theme.$body-font-family;
    font-weight: 700;
    text-align: left;

    > span {
      float: right;
      color: rgba(variables.$white, 0.7);
      font-weight: normal;
    }
  }

  input {
    display: flex;
    flex-shrink: 0;
    justify-content: center;
    align-items: center;
    width: 20px;
    height: 20px;
    margin: 0;
    border-radius: 2px;
    cursor: pointer;
    transition: all 0.5s;
    appearance: none;

    &::before {
      display: inline-block;
      font-family: system-ui;
      font-weight: 900;
      font-size: 0;
      content: 'L';
    }

    &:hover {
      transform: scale(1.1);
    }

    &:not(:checked) {
      border: 2px solid theme.$input-resting-border-color;
      &:hover {
        border-color: theme.$input-hover-border-color;
      }
    }

    &:checked {
      background-color: var(--primary-color);
      &::before {
        font-size: 16px;
        transform: translateY(-1px) scaleX(-1) rotate(-45deg);
      }
    }
  }

  &.error {
    .helperText {
      color: theme.$text-field-error-color;
    }
  }
}

.row {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}

.helperText {
  margin-top: 4px;
  font-family: theme.$body-font-family;
  font-size: 12px;
  text-align: left;
}
