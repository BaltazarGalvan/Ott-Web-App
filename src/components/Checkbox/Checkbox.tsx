import React from 'react';

import useOpaqueId from '../../hooks/useOpaqueId';

import styles from './Checkbox.module.scss';

type Props = {
  label?: string;
  name: string;
  value: boolean;
  onChange: React.ChangeEventHandler<HTMLInputElement>;
  header?: string;
};

const Checkbox: React.FC<Props> = ({ label, name, onChange, header }: Props) => {
  const id = useOpaqueId('check-box', name);

  return (
    <div className={styles.checkbox}>
      {header && <span className={styles.header}>{header}</span>}
      <input name={name} type="checkbox" id={id} onChange={onChange} />
      <label htmlFor={id}>{label}</label>
    </div>
  );
};

export default Checkbox;
