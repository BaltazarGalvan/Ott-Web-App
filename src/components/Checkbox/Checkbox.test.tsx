import React from 'react';
import { render } from '@testing-library/react';

import Checkbox from './Checkbox';

describe('<Checkbox>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Checkbox name="abc" value={false} onChange={(e) => e} />);

    expect(container).toMatchSnapshot();
  });
});
