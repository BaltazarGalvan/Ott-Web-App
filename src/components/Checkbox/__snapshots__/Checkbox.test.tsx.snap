// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox> renders and matches snapshot 1`] = `
<div>
  <div
    class="checkbox"
  >
    <div
      class="row"
    >
      <input
        id="check-box_1235_name"
        name="name"
        type="checkbox"
        value="value"
      />
      <label
        for="check-box_1235_name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;
